addons/api/mirage/factories/auth-method.js

```diff
@@ -10,7 +10,10 @@ export default factory.extend({
   withAccountsAndUsers: trait({
     afterCreate(authMethod, server) {
       const { scope } = authMethod;
-      server.createList('user', 5, { scope }).map(user => {
+      const users = server.createList('user', 5, { scope });
+      // Create authorized user: user123
+      users.push(server.create('user', { scope, id: 'user123' }));
```

Comment:
We can't hardcode the user ID in this way.  This factory could potentially be run on multiple auth methods.  But a given user can only be associated with a single auth method.  What is it you're trying to achieve with this hardcoded user ID?

Reply:
In desktop, list of all sessions need to be filtered down to ones related to logged in user. Seeing as we don't actually create a user for authentication in mirage ('user123' is returned as part user_id of auth payload without any associated user creation in mirage schema), I'm creating the authenticated user and populating sessions data for this user. I was hoping I could inject authenticated user sessions as part of /sessions response in mirage but this request doesn't have access to authenticated user id. There might be some mirage magic to support this scenario. Do you have any pointers on how to do this more cleanly?

Reply:
We don't have to model every detail of the API (and we shouldn't, because we'd recreate the API).  In this case it's probably enough to return any sessions without filtering them by user?

Reply:
Ah. The user filtering happens in ember route action so if we leave things as they are on main, sessions view won't display any data.

Reply:
Updated mock auth payload to use first available user and mock session records to randomly pick between auth user and any user available.

Reply:
Can this be made deterministic somehow?  For example, could it alternate between the two options?  Otherwise it's possible to have no sessions assigned to the logged in user.